app/Services/Model/Imp/TrackChairRankingService.php

```diff
@@ -0,0 +1,228 @@
+
+    private function checkPermissions(Summit $summit, ?Member $member): void
+    {
+        if (is_null($member))
+            throw new AuthzException("User not Found.");
+
+        $isAuth = $summit->isTrackChairAdmin($member);
+
+        if (!$isAuth)
+            throw new AuthzException(sprintf("User %s is not authorized to perform this action.", $member->getId()));
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getTrackChairRatingType(SelectionPlan $selection_plan, int $track_chair_rating_type_id): PresentationTrackChairRatingType
+    {
+        return $this->tx_service->transaction(function () use ($selection_plan, $track_chair_rating_type_id)
+        {
+            $current_member = $this->resource_server_ctx->getCurrentUser();
+            $this->checkPermissions($selection_plan->getSummit(), $current_member);
+
+            $track_chair_rating_type = $selection_plan->getTrackChairRatingTypeById($track_chair_rating_type_id);
+            if (is_null($track_chair_rating_type))
+                throw new EntityNotFoundException("Track chair rating type not found.");
+
+            return $track_chair_rating_type;
+        });
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function addTrackChairRatingType(SelectionPlan $selection_plan, array $payload): PresentationTrackChairRatingType
+    {
+        return $this->tx_service->transaction(function () use ($selection_plan, $payload)
+        {
+            $current_member = $this->resource_server_ctx->getCurrentUser();
+            $this->checkPermissions($selection_plan->getSummit(), $current_member);
+
+            $track_chair_rating_type = new PresentationTrackChairRatingType();
+            $track_chair_rating_type->setWeight(floatval($payload['weight']));
+            $track_chair_rating_type->setName($payload['name']);
+            $track_chair_rating_type->setOrder(intval($payload['order']));
+
+            $selection_plan->addTrackChairRatingType($track_chair_rating_type);
+
+            return $track_chair_rating_type;
+        });
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function updateTrackChairRatingType(
+        SelectionPlan $selection_plan, int $track_chair_rating_type_id, array $payload): PresentationTrackChairRatingType
+    {
+        return $this->tx_service->transaction(function () use ($selection_plan, $track_chair_rating_type_id, $payload)
+        {
+            $current_member = $this->resource_server_ctx->getCurrentUser();
+            $this->checkPermissions($selection_plan->getSummit(), $current_member);
+
+            $track_chair_rating_type = $selection_plan->getTrackChairRatingTypeById($track_chair_rating_type_id);
+            if (is_null($track_chair_rating_type))
+                throw new EntityNotFoundException("Track chair rating type not found.");
+
+            $track_chair_rating_type->setWeight(floatval($payload['weight']));
+            $track_chair_rating_type->setName($payload['name']);
```

Comment:
add validation per name ( 2 raking cant have the same name) if so throw validation exception

Reply:
validations added in rating type and score type "ADD" methods